version: '3.8'

services:
  db:
    image: postgres:15
    container_name: db
    volumes:
      - pg_blog_data:/var/lib/postgresql/data/
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
      - POSTGRES_HOST=${POSTGRES_HOST}
    expose:
      - "5432"
    networks:
      - djangonetwork
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"]
      interval: 2s
      timeout: 1s
      retries: 5
  web:
    build: ../
    tty: true
    container_name: django
    volumes:
      - ../:/usr/src/app/
    ports:
      - 8000:8000
    depends_on:
      db:
        condition: service_healthy
    env_file: "../.env"
    networks:
      - djangonetwork
volumes:
  pg_blog_data:
networks:
  djangonetwork:
      driver: bridge